meson.build

```diff
@@ -48,7 +48,24 @@ if enable_nls
 endif
 
 conf.set('HAVE_LOCALE_T', enable_locale and cc.has_header_symbol('locale.h', 'locale_t'))
-conf.set('HAVE_ICONV', enable_locale and cc.has_function('iconv'))
+iconv_dep = dependency('', required: false)
+if enable_locale
+  iconv_open_snippet = '''#include <iconv.h>
+      int main() {
+      iconv_open("","");
+      }'''
```

Comment:
It is confusing that this variable is defined in a code path where it is not used